pkg/cache/snapshot_test.go

```diff
@@ -40,149 +39,208 @@ func TestSnapshot(t *testing.T) {
 	if err := kueue.AddToScheme(scheme); err != nil {
 		t.Fatalf("Failed adding kueue scheme: %s", err)
 	}
-	cache := New(fake.NewClientBuilder().WithScheme(scheme).Build())
-	clusterQueues := []kueue.ClusterQueue{
+
+	cases := []struct {
+		name          string
+		clusterQueues []*kueue.ClusterQueue
+		flavors       []kueue.ResourceFlavor
+		workloads     []*kueue.Workload
+		wantSnapshot  func(workloads []*kueue.Workload) Snapshot
```

Comment:
```suggestion
		wantSnapshot  Snapshot
```

Avoid building the desired results of the test.